src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -1069,7 +1485,99 @@ private function step_in_head(): bool {
 	 * @return bool Whether an element was found.
 	 */
 	private function step_in_head_noscript(): bool {
-		$this->bail( 'No support for parsing in the ' . WP_HTML_Processor_State::INSERTION_MODE_IN_HEAD_NOSCRIPT . ' state.' );
+		$token_name = $this->get_token_name();
+		$token_type = $this->get_token_type();
+		$is_closer  = parent::is_tag_closer();
+		$op_sigil   = '#tag' === $token_type ? ( $is_closer ? '-' : '+' ) : '';
+		$op         = "{$op_sigil}{$token_name}";
+
+		switch ( $op ) {
+			/*
+			 * > A DOCTYPE token
+			 */
+			case 'html':
+				// Parse error: ignore the token.
+				return $this->step();
+
+			/*
+			 * > A start tag whose tag name is "html"
+			 */
+			case '+HTML':
+				return $this->step_in_body();
+
+			/*
+			 * > An end tag whose tag name is "noscript"
+			 */
+			case '-NOSCRIPT':
+				$this->state->stack_of_open_elements->pop();
+				$this->state->insertion_mode = WP_HTML_Processor_State::INSERTION_MODE_IN_HEAD;
+				return true;
+		}
+
+		/*
+		 * > A character token that is one of U+0009 CHARACTER TABULATION,
+		 * > U+000A LINE FEED (LF), U+000C FORM FEED (FF),
+		 * > U+000D CARRIAGE RETURN (CR), or U+0020 SPACE
+		 *
+		 * Parse error: ignore the token.
+		 */
+		if ( '#text' === $op ) {
+			$text = $this->get_modifiable_text();
+			if ( strlen( $text ) === strspn( $text, " \t\n\f\r" ) ) {
+				return $this->step_in_head();
+			}
+		}
+
+		$is_excluded_closing_tag = false;
+
+		switch ( $op ) {
+			/*
+			 * > A comment token
+			 * >
+			 * > A start tag whose tag name is one of: "basefont", "bgsound",
+			 * > "link", "meta", "noframes", "style"
+			 */
+			case '#comment':
+			case '#funky-comment':
+			case '#presumptuous-tag':
+			case '+BASEFONT':
+			case '+BGSOUND':
+			case '+LINK':
+			case '+META':
+			case '+NOFRAMES':
+			case '+STYLE':
+				return $this->step_in_head();
+
+			/*
+			 * > An end tag whose tag name is "br"
+			 *
+			 * This should never happen, as the Tag Processor prevents showing a BR closing tag.
+			 */
+			case '-BR':
+				/*
+				 * > Act as described in the "anything else" entry below.
+				 */
+				$is_excluded_closing_tag = true;
+				break;
```

Comment:
I didn't understand why `</br>` would get special handling here, but it's to prevent the end tag from being ignored. `</br>` has special handling to be treated as an opener.

It is handled correctly in this PR 👍